tests/integration/test_interaction_assessment.py

```diff
@@ -150,6 +160,86 @@ def test_max_attempts_reached_submit_and_reset_disabled(self):
         self.assertEqual(submit_button.get_attribute('disabled'), 'true')
         self.assertEqual(reset_button.get_attribute('disabled'), 'true')
 
+    def test_show_answer(self):
+        """
+        Test "Show Answer" button is shown in assessment mode, enabled when no
+        more attempts remaining, and displays correct answers when clicked.
+        """
+        show_answer_button = self._get_show_answer_button()
+        self.assertTrue(show_answer_button.is_displayed())
+
+        self.place_item(0, TOP_ZONE_ID)
+        for _ in xrange(self.MAX_ATTEMPTS):
+            self.assertEqual(show_answer_button.get_attribute('disabled'), 'true')
+            self.click_submit()
+
+        self.assertIsNone(show_answer_button.get_attribute('disabled'))
+        self.click_show_answer()
+
+        zones = dict(self.all_zones)
+        for item in self._get_items_with_zone(self.items_map).values():
+            zone_titles = [zones[zone_id] for zone_id in item.zone_ids]
+            # When showing answers, correct items are placed as if assessment_mode=False
+            self.assert_placed_item(item.item_id, zone_titles, assessment_mode=False)
+
+        for item in self._get_items_without_zone(self.items_map).values():
+            self.assertNotDraggable(item.item_id)
+            element = self._get_item_by_value(item.item_id)
+            self.assertEqual(element.get_attribute('aria-grabbed'), 'false')
+            self.assertEqual(element.get_attribute('class'), 'option fade')
+
+            item_content = element.find_element_by_css_selector('.item-content')
+            item_description_id = '-item-{}-description'.format(item.item_id)
+            self.assertEqual(item_content.get_attribute('aria-describedby'), item_description_id)
+
+            describedby_text = (u'Press "Enter", "Space", "Ctrl-m", or "⌘-m" on an item to select it for dropping, '
+                                'then navigate to the zone you want to drop it on.')
+            self.assertEqual(element.find_element_by_css_selector('.sr').text, describedby_text)
+
+    def test_hide_answer(self):
+        """
+        Test "Hide Answer" button is shown correctly, and clicking it returns
+        items to the submitted state.
+        """
+        show_answer_button = self._get_show_answer_button()
+        self.assertEqual(show_answer_button.text, "Show Answer")
+
+        # Place one item correctly
+        self.place_item(0, TOP_ZONE_ID)
+
+        for _ in xrange(self.MAX_ATTEMPTS):
+            self.click_submit()
+
+        # Precondition: check item placement
+        item_description = self._get_placed_item_by_value(0).find_element_by_css_selector('.sr').text
+        self.assertEqual(item_description, 'Placed in: {}'.format(TOP_ZONE_TITLE))
```

Comment:
@arbrandes This logic is reused twice below. You could put it into a nested `check_item_placement` function to DRY out this test.